### PR TITLE
Hide URL bar on scroll (closes #17)

### LINKED ISSUE #17: URL bar should hide when scrolling down down page content

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@etiennesegonzac r?

This was the best was I could think of handling this. Open to alternative ideas :)

**Comment by etiennesegonzac** [comment]
I can scroll the top bar of every tab out while in the tabview.
Which is cool but probably a bug :)

**Comment by etiennesegonzac** [comment]
In some cases the physics is off because of the scroll propagation. Maybe we need to fix #15 first?
But I won't block on it if we fix the tabsview issue :)

And I like this simple reflow-free approach too!

**Comment by etiennesegonzac** [comment]
Maybe a snappoint would help too.

**Comment by wilsonpage** [comment]
> But I won't block on it if we fix the tabsview issue

Which issue?

@etiennesegonzac r?

**Comment by etiennesegonzac** [comment]
> Which issue?

The one where I can scroll tabs while in tabs view mode (I can scroll the previews).

**Comment by wilsonpage** [comment]
> The one where I can scroll tabs while in tabs view mode (I can scroll the previews).

Ah yes, that is fixed now.

**Comment by etiennesegonzac** [comment]
Cool, I'll review soon!

File: lib/hope-tab.js [hunk 1]
```diff
@@ -63,6 +66,43 @@ window.HopeTab = fxosComponent.register('hope-tab', {
     this.url = url;
   },
 
+  onScroll(e) {
```
Comment (etiennesegonzac) [inline lib/hope-tab.js]: nit: onBrowserScroll

(I'm afraid we'll start to get confused at some point :))

File: lib/hope-tab.js [hunk 2]
```diff
@@ -115,11 +155,18 @@ window.HopeTab = fxosComponent.register('hope-tab', {
...
+    clearTimeout(this.showBarTimeout);
+    this.els.inner.style.overflowY = 'scroll';
   },
 
   deactivate() {
     this.removeAttr('active');
     this.els.iframe.setVisible(false);
+    this.showBar();
+
+    this.showBarTimeout = setTimeout(() => {
```
Comment (etiennesegonzac) [inline lib/hope-tab.js]: nit: should maybe be a scheduler mutation block

**Comment by wilsonpage** [comment]
@etiennesegonzac I found the scroll-snap was interfering with `.scrollTo()` so have removed for the time being. Latest commit seems more solid, being smarter instead of hammering the `.scrollTo()` API.

r?

File: lib/hope-tab.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*global fxosComponent, scheduler */
 (function() {
 
-var debug = 0 ? (...args) => console.log('[hope-tab]', ...args) : () => {};
+var debug = 1 ? (...args) => console.log('[hope-tab]', ...args) : () => {};
```
Comment (etiennesegonzac) [inline lib/hope-tab.js]: let's not forget the debug statement

File: lib/hope-tab.js [hunk 4]
```diff
@@ -115,11 +167,18 @@ window.HopeTab = fxosComponent.register('hope-tab', {
   activate() {
     this.setAttr('active', '');
     this.els.iframe.setVisible(true);
+    clearTimeout(this.showBarTimeout);
+    this.els.inner.style.overflowY = 'scroll';
```
Comment (etiennesegonzac) [inline lib/hope-tab.js]: this is still a mutation :)

**Comment by etiennesegonzac** [comment]
Don't know if there's something wrong with my setup but I can't select a different tab anymore :/
And when entering the tabsview the url bar of the current tab doesn't appear.

**Comment by etiennesegonzac** [comment]
ha, that explains it :) `TypeError: this.showBar is not a function"`

File: lib/hope-tab.js (hunk 4 above)
Comment (wilsonpage) [inline lib/hope-tab.js]: I just noticed that this add a noticeable delay when open in scrolling to the current tab. As the `.mutation()` task is blocked until scrolling has finished. Previously the overlay would be faded out  while scrolling was still in progress.

**Comment by wilsonpage** [comment]
@etiennesegonzac r?

**Comment by etiennesegonzac** [comment]
Alright, ready to squash and land :)

One thing that I'm not sure I completely understand is why the `.overlay` isn't preventing the scrolling by itself (causing us to to pretty big reflows to change the overflowY), but we can investigate in a follow up.

**Comment by wilsonpage** [comment]
> One thing that I'm not sure I completely understand is why the .overlay isn't preventing the scrolling by itself (causing us to to pretty big reflows to change the overflowY), but we can investigate in a follow up.

This could work if overlay was a sibling to a scrollable container that contained `.bar` and `.content`.

**Comment by wilsonpage** [comment]
@etiennesegonzac squashing and landing.

**Pull request merged**